admin_terraform_version.go

```diff
@@ -0,0 +1,176 @@
+type AdminTerraformVersion struct {
+	ID        string    `jsonapi:"primary,terraform-versions"`
+	Version   string    `jsonapi:"attr,version"`
+	URL       string    `jsonapi:"attr,url"`
+	Sha       string    `jsonapi:"attr,sha"`
+	Official  bool      `jsonapi:"attr,official"`
+	Enabled   bool      `jsonapi:"attr,enabled"`
+	Beta      bool      `jsonapi:"attr,beta"`
+	Usage     int       `jsonapi:"attr,usage"`
+	CreatedAt time.Time `jsonapi:"attr,created-at,iso8601"`
+}
+
+// AdminTerraformVersionsListOptions represents the options for listing
+// terraform versions.
+type AdminTerraformVersionsListOptions struct {
+	ListOptions
+}
+
+// AdminTerraformVersionsList represents a list of terraform versions.
+type AdminTerraformVersionsList struct {
+	*Pagination
+	Items []*AdminTerraformVersion
+}
+
+// List all the terraform versions.
+func (a *adminTerraformVersions) List(ctx context.Context, options AdminTerraformVersionsListOptions) (*AdminTerraformVersionsList, error) {
+	req, err := a.client.newRequest("GET", "admin/terraform-versions", &options)
+	if err != nil {
+		return nil, err
+	}
+
+	tvl := &AdminTerraformVersionsList{}
+	err = a.client.do(ctx, req, tvl)
+	if err != nil {
+		return nil, err
+	}
+
+	return tvl, nil
+}
+
+// Read a terraform version by its ID.
+func (a *adminTerraformVersions) Read(ctx context.Context, id string) (*AdminTerraformVersion, error) {
+	if !validStringID(&id) {
+		return nil, ErrInvalidTerraformVersionID
+	}
+
+	u := fmt.Sprintf("admin/terraform-versions/%s", url.QueryEscape(id))
+	req, err := a.client.newRequest("GET", u, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	tfv := &AdminTerraformVersion{}
+	err = a.client.do(ctx, req, tfv)
+	if err != nil {
+		return nil, err
+	}
+
+	return tfv, nil
+}
+
+// AdminTerraformVersionCreateOptions for creating a terraform version.
+// https://www.terraform.io/docs/cloud/api/admin/terraform-versions.html#request-body
+type AdminTerraformVersionCreateOptions struct {
+	Type     string  `jsonapi:"primary,terraform-versions"`
+	Version  *string `jsonapi:"attr,version"`
+	URL      *string `jsonapi:"attr,url"`
+	Sha      *string `jsonapi:"attr,sha"`
+	Official *bool   `jsonapi:"attr,official"`
+	Enabled  *bool   `jsonapi:"attr,enabled"`
+	Beta     *bool   `jsonapi:"attr,beta"`
+}
+
+// Create a new terraform version.
+func (a *adminTerraformVersions) Create(ctx context.Context, options AdminTerraformVersionCreateOptions) (*AdminTerraformVersion, error) {
+	req, err := a.client.newRequest("POST", "admin/terraform-versions", &options)
+	if err != nil {
+		return nil, err
+	}
+
+	tfv := &AdminTerraformVersion{}
+	err = a.client.do(ctx, req, tfv)
+	if err != nil {
+		return nil, err
+	}
+
+	return tfv, nil
+}
+
+// AdminTerraformVersionUpdateOptions for updating terraform version.
+// https://www.terraform.io/docs/cloud/api/admin/terraform-versions.html#request-body
+type AdminTerraformVersionUpdateOptions struct {
+	Type     string  `jsonapi:"primary,terraform-versions"`
```

Comment:
I really like what you landed on here, with `Type` used in `*Options` but leaving the necessary tag as `ID` in the struct (since jsonapi does say it [should be used on the primary key](https://github.com/google/jsonapi#primary)) Nice work!

One thing I'd consider adding in your followup is a docstring indicating what this is for in each `*Options` (something far more helpful than "Internal use only!"); maybe even just `See /link/to/readme` 🤷

Reply:
😄 .

> adding in your followup is a docstring indicating what this is for

👍 will do